### PR TITLE
Elasticsearch: Fix impossibility to perform non-logs queries after importing queries from loki or prometheus in explore

### PR DESCRIPTION
**Pull request by Elfo404**
When importing queries in explore we were setting `isLogsQuery: true` which made it impossible to perform non-logs queries in elasticsearch when switching from  a loki/prometheus datasource.

`isLogsQuery` is effectively useless since we can check for the existence of the "virtual" logs aggregation to determine if a query is a logs query.

**Note:** I'm not sure how to better phrase the PR title, feel free to change it if you have a better idea.

### CODE REVIEW

File: public/app/plugins/datasource/elasticsearch/specs/elastic_response.test.ts [hunk 1]
```diff
@@ -1208,17 +1208,12 @@ describe('ElasticResponse', () => {
...
       {
         refId: 'A',
         metrics: [{ type: 'count', id: '1' }],
         bucketAggs: [{ type: 'date_histogram', settings: { interval: 'auto' }, id: '2' }],
-        context: 'explore',
-        interval: '10s',
-        isLogsQuery: true,
         key: 'Q-1561369883389-0.7611823271062786-0',
-        liveStreaming: false,
-        maxDataPoints: 1620,
```
Comment (Elfo404) [inline public/app/plugins/datasource/elasticsearch/specs/elastic_response.test.ts:1221]: not sure why those were here as they are not part of the ES query model.

File: public/app/plugins/datasource/elasticsearch/query_def.ts [hunk 2]
```diff
@@ -42,8 +43,8 @@ export function defaultBucketAgg(id = '1'): BucketAggregation {
 export const findMetricById = (metrics: MetricAggregation[], id: MetricAggregation['id']) =>
   metrics.find((metric) => metric.id === id);
 
-export function hasMetricOfType(target: any, type: string): boolean {
-  return target && target.metrics && target.metrics.some((m: any) => m.type === type);
+export function hasMetricOfType(target: ElasticsearchQuery, type: string): boolean {
```
Comment (Elfo404) [inline public/app/plugins/datasource/elasticsearch/query_def.ts:46]: `type: string` could be instead `type: MetricAggregationType`

File: public/app/plugins/datasource/elasticsearch/datasource.ts [hunk 3]
```diff
@@ -583,7 +583,8 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
       map((res) => {
         const er = new ElasticResponse(sentTargets, res);
 
-        if (sentTargets.some((target) => target.isLogsQuery)) {
+        // TODO: This needs to be revisited, it seems wroong to process ALL the sent queries as logs if only one of them was a log query
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/elasticsearch/datasource.ts:586]:
```suggestion
        // TODO: This needs to be revisited, it seems wrong to process ALL the sent queries as logs if only one of them was a log query
```

**Review by ivanahuckova — APPROVED** [review]
LGTM! I have also checked http://localhost:3000/d/M94gguRWz/datasource-tests-elasticsearch-v7-filebeat?orgId=1 to see if logs queries work as expected. And all 👀  good.

File: public/app/plugins/datasource/elasticsearch/datasource.ts (hunk 3 above)
Comment (Elfo404) [reply public/app/plugins/datasource/elasticsearch/datasource.ts:586]: I assume it really felt wrooooooong in my mind 😅 thanks for catching it :)

**Pull request merged**